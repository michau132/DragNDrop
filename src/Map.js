import React from 'react';
import styled from 'styled-components';
import {
  Map as LeafletMap, TileLayer, Marker, Popup,
} from 'react-leaflet';
import PropTypes from 'prop-types';

const StyledLeafletMap = styled(LeafletMap)`
  height: 300px;
`;

class Map extends React.Component {
  static propTypes = {
    images: PropTypes.arrayOf(
      PropTypes.shape({
        position: PropTypes.shape({
          lat: PropTypes.number.isRequired,
          lng: PropTypes.number.isRequired,
        }).isRequired,
      }).isRequired,
    ).isRequired,
  }

  constructor(props, context) {
    super(props, context);
    this.mapApi = React.createRef();
  }

  render() {
    const { images } = this.props;
    return (
      <StyledLeafletMap center={[45, 25]} zoom={4} ref={this.mapApi}>
        <TileLayer
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"
        />
        {
          images.map((marker) => {
            const position = [marker.position.lat, marker.position.lng];
            return (
              <Marker position={position} key={marker.avatar}>
                <Popup>{marker.name}</Popup>
              </Marker>
            );
          })
        }
      </StyledLeafletMap>
    );
  }
}

export default Map;
