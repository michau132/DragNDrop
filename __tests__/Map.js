import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Map from '../src/Map';

const props = {
  images: [
    {
      avatar: 'someUrl',
      position: {
        lng: 12.12,
        lat: 12.13,
      },
      name: 'fileName.jpg',
    },
    {
      avatar: 'secondUrl',
      position: {
        lng: 13,
        lat: 50,
      },
      name: 'DESC414.jpg',
    },
  ],
};

describe('<Map/> component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Map {...props} />);
  });
  test('should be defined', () => {
    expect(wrapper).toBeDefined();
  });
  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
