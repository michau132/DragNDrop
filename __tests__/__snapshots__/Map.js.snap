// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map/> component should match snapshot 1`] = `
<Styled(Map)
  center={
    Array [
      45,
      25,
    ]
  }
  zoom={4}
>
  <ForwardRef(Leaflet(TileLayer))
    attribution="Â© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
    url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"
  />
  <ForwardRef(Leaflet(Marker))
    key="someUrl"
    position={
      Array [
        12.13,
        12.12,
      ]
    }
  >
    <ForwardRef(Leaflet(Popup))>
      fileName.jpg
    </ForwardRef(Leaflet(Popup))>
  </ForwardRef(Leaflet(Marker))>
  <ForwardRef(Leaflet(Marker))
    key="secondUrl"
    position={
      Array [
        50,
        13,
      ]
    }
  >
    <ForwardRef(Leaflet(Popup))>
      DESC414.jpg
    </ForwardRef(Leaflet(Popup))>
  </ForwardRef(Leaflet(Marker))>
</Styled(Map)>
`;
